software/delly/call/meta.yml

```diff
@@ -0,0 +1,55 @@
+name: delly_call
+description: Call structural variants
+keywords:
+  - genome
+  - structural
+  - variants
+  - bcf
+
+tools:
+  - delly:
+      description: Structural variant discovery by integrated paired-end and split-read analysis
+      homepage: https://github.com/dellytools/delly
+      documentation: https://github.com/dellytools/delly/blob/master/README.md
+      tool_dev_url: None
+      doi: "DOI:10.1093/bioinformatics/bts378"
+      licence: ["BSD-3-clause"]
+
+input:
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - bam:
+      type: file
+      description: BAM file from alignment must be sorted, indexed, and duplicate marked
```

Comment:
Does the BAM have to be duplicate marked?

Reply:
@drpatelh Yeah and I tried to use the test file that's duplicate marked but the chr names between the fasta and bam didn't match which caused a problem for delly. Would this be fine? If not, I might need to upload the test data where both fasta + bam has matching sequence names.

```
head data_modules/data/genomics/homo_sapiens/genome/genome.fasta
>chr22
```

```
samtools view -H data_modules/data/genomics/homo_sapiens/illumina/bam/test.paired_end.markduplicates.sorted.bam
@HD	VN:1.6	SO:coordinate
@SQ	SN:chr22:16570000-16610000	LN:40001
@RG	ID:1	PU:1	SM:testN	LB:testN	PL:illumina
```

Reply:
Cool. Those chromosome names look the same? `chr22` in both?

Reply:
It seems that delly requires the sequence names to be the same; so in both files it should be: `chr22:16570000-16610000`

Reply:
Ahhhh, I see. I wonder how that BAM was generated then? 🤔

What does the output of the following command look like? Want to see what the chromosome names in the alignment look like.

```
samtools view data_modules/data/genomics/homo_sapiens/illumina/bam/test.paired_end.markduplicates.sorted.bam | head
```

Reply:
```
samtools view data_modules/data/genomics/homo_sapiens/illumina/bam/test.paired_end.markduplicates.sorted.bam | head
testN:0	419	chr22:16570000-16610000	1917	60	40M103H	=	2007	194	AGCAATGTAAAATTTATTGAAAATGTATTTGCTTTTTNTG	CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC!CC	SA:Z:chr22:16570000-16610000,2007,+,90S53M,60,0;	MC:Z:104M	MD:Z:11G25T2	PG:Z:MarkDuplicates	RG:Z:1	NM:i:2	AS:i:33	XS:i:0
testN:1	163	chr22:16570000-16610000	1952	60	130M13S	=	2014	157	TTGTGTAAATCATCTGTGAATCCAGAGGGGAAAAATATGACAAAGAAAGCTATATAAGATATTATTTTATTTTACAGAGTAACAGACTAGCTAGAGACAATGAATTAAGGGAAAATGACAAAGAACAGCTAGTCTGTTACTCT	.CCCCCC@.CCCCC@@CCCCCC@@CCCC..@CCCCCC.CCCCCCC@CCCC.CC...C@CCC.CC.CCC@C.CCC.C;@CCCC@@CCC@CC.CCCCCC.CCCC;@@CCC5.@@.@C.C;@;CCC@C...5@@C...C.C@;;;5	MC:Z:48S95M	MD:Z:2T127	PG:Z:MarkDuplicates	RG:Z:1	NM:i:1	AS:i:127	XS:i:0
testN:4	163	chr22:16570000-16610000	1959	60	143M	=	1972	156	AATTATCTGTGAATCGAGAGGGGAAAAATATGACAAAGAAAGCTATATAAGATATTATTTGGTTTTACAGAGTAACATACTAGCTAGAGACAATGAATTAAGGGAAAATGACAAAGAACAGCTCAAAGCAATTTCTACACGAG	CCCCCCCCCCCCCCC.CCCCCCCCC@CCCCCCCCCCCCCCCCCC@CCCCC;CCCCC@CCC@CCCCCCCCCCCCCCCCC.@CCCCCCCCCCC.CC;CCCCCCCCCCCCCC@C5CCC@.CCCC@@;@C@C@CCCCC@CC@C@@@;	MC:Z:143M	MD:Z:3C11C44T0A15G65	PG:Z:MarkDuplicates	RG:Z:1	NM:i:5	AS:i:119	XS:i:0
testN:5	163	chr22:16570000-16610000	1963	60	135M	=	1963	135	ATCTGTGAATCCAGAGGGGAAAAATATGACAAAGAAAGCTATATAAGATATTATTTTATTTTACAGAGTAACAGACTAGCTAGAGACAATGAATTAAGGGAAAATGACAAAGAACAGCTCAAAGCAATTTCTACA	NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN	MC:Z:135M	MD:Z:135	PG:Z:MarkDuplicates	RG:Z:1	NM:i:0	AS:i:135	XS:i:0
testN:6	1187	chr22:16570000-16610000	1963	60	135M	=	1963	135	ATCTGTGAATCCAGAGGGGAAAAATATGACAAAGAAAGCTATATAAGATATTATTTTATTTTACAGAGTAACAGACTAGCTAGAGACAATGAATTAAGGGAAAATGACAAAGAACAGCTCAAAGCAATTTCTACA	@CCCCCC;CCCCCCCC@CCCCCCCCCC@CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC@CCCCCC@CCCCCCCCC;@;C;;CCCCCCCCCCC;@@CCC@@@5@CC@C;@C;@@CCCCC@@CCCCC@C..	MC:Z:135M	MD:Z:135	PG:Z:MarkDuplicates	RG:Z:1	NM:i:0	AS:i:135	XS:i:0
testN:7	163	chr22:16570000-16610000	1963	60	143M	=	1971	151	ATCTGTGAATCCAGAGGGGAAAAATATGACAAAGAAAGCTATATAAGATATTATTTTATTTTACAGAGTAACAGACTAGCTAGAGACAATGAATTAAGGGAAAATGACAAAGAACAGCTCAAAGCAATTTCTACACGAGATCC	NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN	MC:Z:143M	MD:Z:143	PG:Z:MarkDuplicates	RG:Z:1	NM:i:0	AS:i:143	XS:i:0
testN:8	1123	chr22:16570000-16610000	1963	60	143M	=	1971	151	ATCTGTGAATCCAGAGGGGAAAAATATGACAAAGAAAGCTATATAAGATATTATTTTATTTTACAGAGTAACAGACTAGCTAGAGACAATGAATTAAGGGAAAATGACAAAGAACAGCTCAAAGCAATTTCTACACGAGATCC	CCCCCCCCCCCCCCCCCCCCCCCCCCCCC@CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC@CCCCC@C@@CCCCCCCCCCCC@CC;.@CCCCCCCCCCCC@CCCC@CCCCCCCCCC@CCCCC;CCC@CC@@C@CC.C	MC:Z:143M	MD:Z:143	PG:Z:MarkDuplicates	RG:Z:1	NM:i:0	AS:i:143	XS:i:0
testN:5	83	chr22:16570000-16610000	1963	60	135M	=	1963	-135	ATCTGTGAATCCAGAGGGGAAAAATATGACAAAGAAAGCTATATAAGATATTATTTTATTTTACAGAGTAACAGACTAGCTAGAGACAATGAATTAAGGGAAAATGACAAAGAACAGCTCAAAGCAATTTCTACA	NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN	MC:Z:135M	MD:Z:135	PG:Z:MarkDuplicates	RG:Z:1	NM:i:0	AS:i:135	XS:i:0
testN:6	1107	chr22:16570000-16610000	1963	60	135M	=	1963	-135	ATCTGTGAATCGAGAGTGGAAATATATGACAAAGAAAGCTATATAAGATATTATTTTATTTTACAGTGTAACAGACTAGCTAGAGACAATGAATTAAGGGAAAATGACAATGATCAGCTCAAAGCAATTTCTACT	5;@C@;..5;..@.5;.@;..;..C.@CC;.;.;CC.C.;..C...;5;@.@C@@...@.@CC..@.@.;C.;@C.C@C.CCC.CC5CCCC@..CCC.C@C.C.@C@C.@.C@.CCC@CC...C.CC@.C5CC..	MC:Z:135M	MD:Z:11C4G5A43A43A2A20A0	PG:Z:MarkDuplicates	RG:Z:1	NM:i:7	AS:i:104	XS:i:0
testN:9	99	chr22:16570000-16610000	1964	60	143M	=	1972	151	TCTGTGAATCCAGAGGGGAAAAATATGACAAAGAAAGCTATATAAGATATTATTTTATTTTACAGAGTAACAGACTAGCTAGAGACAATGAATTAAGGGAAAATGACAAAGAACAGCTCAAAGCAATTTCTACACGAGATCCT	NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNMMNNNNNNMNNNNMNNNNNNNMNNNNNNNNNNNNNNNNNNNMNMNNNNMNNNNNNNNNMNNMMMNNMNMNNNMMNNNNNNMNMMMNNNNMMMMNMMNNNM	MC:Z:143M	MD:Z:143	PG:Z:MarkDuplicates	RG:Z:1	NM:i:0	AS:i:143	XS:i:0
```

Reply:
@FriederikeHanssen any ideas why the chromosome names in these BAM files look like intervals rather than just the chromosome name i.e. `chr22`?

Reply:
Hm no. I took the reads and ran them through sarek and took that output. Maxime also reported some issues with the test data :( I should have some time later today or tomorrow morning to look through it.

Reply:
Thank you! Maybe worth tracking back to an intermediate BAM file that looked ok and figuring out which commands were run to get this one? Could be an artifact of downsampling the genome with the smaller intervals somehow.

Reply:
hm ok, so if I understand it correctly, than the naming difference is the main problem here. I am wondering, if some name changes sneaked in when I tried fixing the bed files and what not. I'll rerun sarek on the small data and check the bam files for the naming. Worst case, have to replace them and fix the checksums in the remaining modules